Project folder restructure, added Token generation from backend

mrvanagas opened the pull request:
Restructured project folder to separate client and server folders. Added server to generate a token. At the moment, token is generated by user click of a button as show in the picture:
![image](https://user-images.githubusercontent.com/71028004/165469474-1c8040da-6a90-42ee-b541-110c7bf60881.png)

mrvanagas committed:
basic api functions

Added 3 basic API functions, also gets local coordinates

mrvanagas committed:
basic API functions

Basic API functions with 3 calls

mrvanagas committed:
Project restructure, added server

Restructured project structure to separate folders, added backend server to generate tokens.

JuliaShavyrova commented on client/package.json:
```diff
@@ -20,6 +20,8 @@
     "axios": "0.21.4",
     "core-js": "3.2.1",
+    "dotenv": "^16.0.0",
+    "dotenv-webpack": "^7.1.0",
```
do you need both packages?

mrvanagas replied:
No, one of them will be removed, once I figure out which one I need.

mrvanagas replied:
Got rid of dotenv dependancy

JuliaShavyrova commented on client/src/API/api.js:
```diff
@@ -0,0 +1,35 @@
+});
+
+const lon = '23.9036';
+const lat = '54.8985';
```
no need to store fixed position

mrvanagas replied:
yes, this is for testing purposes. The idea for the first iteration is for the app to automatically get the user location and show the weather info according to the coordinates.

JuliaShavyrova replied:
please make it work without hardcoded values, it's much easier to do it now then later with more code added

mrvanagas replied:
Fixed hard coded values, now coordinates are given from getLocation() function

JuliaShavyrova commented on client/src/Utils/geoLocator.js:
```diff
@@ -0,0 +1,9 @@
+  console.error(error);
+};
+
+navigator.geolocation.getCurrentPosition(successCallback, errorCallback);
```
don't see any `export` in the file, but it actually imported in `App` component. just interested - does it work?

mrvanagas replied:
Yes, GeoLocation is called automatically once the app is loaded:
![image](https://user-images.githubusercontent.com/71028004/165476183-daa9c618-ec17-4c45-aa53-e014362cd411.png)
We can see getLocation is not called in App.jsx, however in browser console we can see that it has been called automatically without any user input:
![image](https://user-images.githubusercontent.com/71028004/165476479-c67940c8-caf3-48e7-8a76-636b1b36a458.png)
Why does it work like that, I have not figured out exactly.

JuliaShavyrova replied:
> Why does it work like that, I have not figured out exactly.

please make your code more predictable or find out why it works that way and can be used here :)

mrvanagas replied:
Reworked the geoLocator.js file, now it activates only when called and it is being called in the API.js file, in order to supply current coordinates of the user to the app.

EDIT: The promise does not pass the coordinate values to the coord variables, investigating now.

mrvanagas replied:
Reworked the logic of geolocator.js file and used hooks to call and use the information.

JuliaShavyrova commented on client/src/configs/token.js:
```diff
@@ -0,0 +1,3 @@
+const token =
+  'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOlwvXC9wZmEuZm9yZWNhLmNvbVwvYXV0aG9yaXplXC90b2tlbiIsImlhdCI6MTY1MDYzNjIwMiwiZXhwIjo5OTk5OTk5OTk5LCJuYmYiOjE2NTA2MzYyMDIsImp0aSI6ImU3ODNiOTA5NDdhZjZlYjciLCJzdWIiOiJsdWtvbmVsaXMxIiwiZm10IjoiWERjT2hqQzQwK0FMamxZVHRqYk9pQT09In0.aZP7m278J_ToDZFqLUVBARHs161WQf-bjhbwa1y_fIo';
```
please don't store hardcoded token in repository

mrvanagas replied:
This will be removed, once I implement silent authorization.

mrvanagas replied:
Silent auth implemented, removed token file

mrvanagas committed:
removed token

Removed hardcoded token

mrvanagas committed:
Adjusted geolocator function

Geolocator now is called automatically in API.js file to get current current coordinates

Commit:
Auto weather info

Added logic to automatically show weather, based on current position, using the geolocator

Commit:
code modification

JuliaShavyrova commented on client/src/Utils/geoLocator.js:
```diff
@@ -0,0 +1,20 @@
+//   });
+// }
+
+//original location function
```
please remove unused code

mrvanagas replied:
Removed unused code

JuliaShavyrova commented on client/src/configs/token.js:
```diff
@@ -0,0 +1,3 @@
+const token =
+  'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOlwvXC9wZmEuZm9yZWNhLmNvbVwvYXBpXC92MVwvXC9hdXRob3JpemVcL3Rva2VuIiwiaWF0IjoxNjUxMjE0NTgxLCJleHAiOjk5OTk5OTk5OTksIm5iZiI6MTY1MTIxNDU4MSwianRpIjoiMTI0ZWNlOTg5MGY5ZDRiOCIsInN1YiI6Imx1a29uZWxpczEiLCJmbXQiOiJYRGNPaGpDNDArQUxqbFlUdGpiT2lBPT0ifQ.yrTZFMKWGHro_ZPBIBRgQ1E8i36quPRZb4NJDCkRkx8';
```
still see hardcoded token, please remove it

mrvanagas replied:
hard coded token removed

JuliaShavyrova replied:
Do you still need this file?

mrvanagas committed:
Removed hardcoded token, added other pages and router

Removed hard coded token, added several pages as placeholders, added router

mrvanagas committed:
Removed token

Removed hardcoded token

mrvanagas committed:
added 1st iteration of silent auth

Added first iteration of silent authentication, exploring other options with timed tokens

mrvanagas committed:
fixed files

fixed wrong file deletion

JuliaShavyrova commented:
please check destination branch

mrvanagas commented:
Done, changed destination branch.
